cdap-docs/build.sh

```diff
@@ -499,47 +370,48 @@ function clean_outer_level() {
 
 function build_license_dependency_pdfs() {
   cd ${SCRIPT_PATH}/reference-manual
-  ./build.sh license-pdfs ${ARG_2} 
+  ./build.sh license-pdfs
 }
   
 function print_version() {
   cd ${SCRIPT_PATH}/developers-manual
-  ./build.sh display-version ${ARG_2} 
+  ./build.sh display-version
 }
 
-function set_display_version() {
+function set_and_display_version() {
   set_version
   display_version
 }
 
-function ring_bell() {
-  # Pass a message as ${1}
-  if [[ "x${BELL}" != "x" ]]; then
-    echo "$(tput bel)${1}"
-  else
-    echo "${1}"
-  fi
-}
-
-function docs_test2() {
-  echo_red_bold "Test2..."
-  echo "NO_JAVADOCS: ${NO_JAVADOCS}"
-  echo "End of Test2 function"
+function set_project_path() {
+  PROJECT_PATH="${SCRIPT_PATH}/.."
+  PROJECT_PATH=$(cd ${PROJECT_PATH} && pwd)
+  SOURCE_PATH="${SCRIPT_PATH}/../../"
+  SOURCE_PATH=$(cd ${SOURCE_PATH} && pwd)
 }
 
-function docs_test() {
-  echo_red_bold "Test..."
-  echo "${WARNING}"
-  ring_bell "A test message"
-  echo "${RED_BOLD}Red bold${NO_COLOR}text"
-  echo "${BOLD}bold${RED}Red${NO_COLOR}text"
-  echo "${RED}Red${NO_COLOR}text"
-  local recover=$(docs_test2)
-  echo "recover: "
-  echo "${recover}"
-  local last_line=$(echo "${recover}" | tail -n1)
-  echo "last_line: ${last_line}"
-  echo "Test completed."
+function setup() {
+  # Check that we're starting in the correct directory
+  local quiet={$1}
+  E_WRONG_DIRECTORY=85
+  if [[ -z ${MANUAL} ]] || [[ -z ${CDAP_DOCS} ]]; then
+    echo "Manual or CDAP_DOCS set incorrectly: are you in the correct directory?"
+    exit ${E_WRONG_DIRECTORY}
+  fi
+  if [[ " ${MANUALS[@]}" =~ "${MANUAL} " || "${MANUAL}" == "${CDAP_DOCS}" ]]; then
+    if [[ -z ${quiet} ]]; then
+      echo "Check for starting directory: Using \"${MANUAL}\""
+    fi
+    set_project_path
+    if [[ -z ${DEBUG} ]]; then
+      DEBUG="${FALSE}"
+    fi
+    return 0
+  else  
+    echo "Did not find MANUAL \"${MANUAL}\": are you in the correct directory?"
+    exit ${E_WRONG_DIRECTORY}
+  fi
+  exit 1
```

Comment:
Functions should never exit (except main), IMO. They should always "return" and it should be up to the caller to determine whether to exit or not.